app/scripts/controllers/serviceInstance.js

```diff
@@ -73,18 +73,40 @@ angular.module('openshiftConsole')
       DataService.unwatchAll(secretWatchers);
       secretWatchers = [];
 
+      $scope.allowParametersReveal = AuthorizationService.canI('secrets', 'get', $scope.projectName);
       $scope.parameterData = {};
-      _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function(key) {
-        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      $scope.opaqueParameterKeys = [];
+
+      if (!$scope.allowParametersReveal) {
+        _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function (key) {
+          $scope.parameterData[key] = '*****';
```

Comment:
I think we should be able to just fill in the parameter data from the status parameters below and just remove this `if` block. Are you seeing parameters in the schema, but not in status?

Reply:
I guess there are parameters that haven't been set. But since users can already see that in the instance YAML under status, I don't think we need to obfuscate it here.

Reply:
Disregard my comment here. Thinking about this more, I think this behavior is better.